crates/ty_python_semantic/src/types/ide_support.rs

```diff
@@ -28,6 +29,21 @@ use rustc_hash::FxHashSet;
 pub use resolve_definition::{ImportAliasResolution, ResolvedDefinition, map_stub_definition};
 use resolve_definition::{find_symbol_in_scope, resolve_definition};
 
+// `__init__`, `__repr__`, `__eq__`, `__ne__` and `__hash__` are always included via `object`,
+// so we don't need to list them here.
+const SYNTHETIC_DATACLASS_ATTRIBUTES: &[&str] = &[
+    "__lt__",
+    "__le__",
+    "__gt__",
+    "__ge__",
+    "__replace__",
+    "__setattr__",
+    "__delattr__",
+    "__slots__",
+    "__weakref__",
```

Comment:
I think there's some subtlety here where this is only added to a _slotted_ dataclass if `weakref_slot=True` in the dataclass decorator? Not hugely important, though I guess you could add a failing test for it.

Is it worth adding `__match_args__` to this list now ("prematurely")? Then the all_members test for `__match_args__` will just automatically start passing when we add support for the synthesised attribute on the type-inference side